dotnet-authserver/src/TeacherIdentity.AuthServer/Controllers/AuthorizationController.cs

```diff
@@ -184,15 +186,17 @@ public async Task<IActionResult> Authorize()
                     }));
             }
 
-            if (authTicketIsTooOld || request.HasPrompt(Prompts.Login))
-            {
-                authenticationState.Reset(DateTime.UtcNow);
-            }
-
             var signInJourney = _signInJourneyProvider.GetSignInJourney(authenticationState, HttpContext);
             return Redirect(signInJourney.GetStartStepUrl());
         }
 
+        if (trnToken is not null)
+        {
+            // If we have a signed in user we may never enter the TRN token sign in journey, so we have to apply
+            // the token here
+            await _trnTokenHelper.ApplyTrnTokenToUser(authenticationState.UserId, trnToken.TrnToken);
```

Comment:
This now looks wrong to me. There are no checks between user Email and Trn Token Email for example